modules/client/src/erc20/internal/client/methods.ts

```diff
@@ -339,15 +347,26 @@ export class ClientErc20Methods extends ClientCore
         erc20VotingProposals.map(
           (
             proposal: SubgraphErc20ProposalListItem,
-          ): Promise<Erc20ProposalListItem> => {
+          ): Promise<Erc20ProposalListItem> | Erc20ProposalListItem => {
             // format in the metadata field
-            return this.ipfs
-              .fetchString(proposal.metadata)
-              .then((stringMetadata: string) => {
-                // TODO: Parse and validate schema¡
-                const metadata = JSON.parse(stringMetadata) as ProposalMetadata;
-                return toErc20ProposalListItem(proposal, metadata);
-              });
+            try {
+              const metadataCid = resolveIpfsCid(proposal.metadata);
+              return this.ipfs
+                .fetchString(metadataCid)
+                .then((stringMetadata: string) => {
+                  // TODO: Parse and validate schema¡
+                  const metadata = JSON.parse(
+                    stringMetadata,
+                  ) as ProposalMetadata;
+                  return toErc20ProposalListItem(proposal, metadata);
+                });
+            } catch (err) {
```

Comment:
Shouldn't this try/catch be inside of the `.then()` clause? 

This try/catch block will only trigger in case the sync code throws an Error, but internal promise-related stuff would be out of scope here

Reply:
Also, IPFS fetching could also fail because the data isn't available. We should make a difference here. Identify if the thrown error is because of the invalid IPFS Link.... otherwise, I would return a metadata object with `title: "(metadata not available)"`